[CI] - Kubernetes end-to-end testing

This issue will be open and closed dependent on the state of Kubernetes end-to-end testing building against Quarkus main snapshot.

If you have interest in being notified of this subscribe to the issue.


<!-- status.quarkus.io/status:
---
updatedAt: 1769828995.087290094
failure: false
repository: "quarkusio/quarkus"
runId: 21537575867
quarkusSha: "a22c735396db4b35015cd28ee7cb9c4696cb8159"
projectSha: "a22c735396db4b35015cd28ee7cb9c4696cb8159"
firstFailure: null
lastFailure: null
lastSuccess:
  date: 1769828994.027101936
  quarkusSha: "a22c735396db4b35015cd28ee7cb9c4696cb8159"
  projectSha: "a22c735396db4b35015cd28ee7cb9c4696cb8159"
-->

I'll try to take a look later, but this seems more related to the content of the resource and not to the API invocation itself (selector doesn't match template labels).
I'm not sure if this is done through Dekorate, or if it's implemented directly in the Quarkus code-base.

Related error log:
```
Error: [INFO] [ERROR] 	[error]: Build step io.quarkus.kubernetes.deployment.KubernetesDeployer#deploy threw an exception: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://10.1.0.191:8443/apis/apps/v1/namespaces/default/statefulsets. Message: StatefulSet.apps "kubernetes-docker-build-and-deploy-statefulset" is invalid: spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`. Received status: Status(apiVersion=v1, code=422, details=StatusDetails(causes=[StatusCause(field=spec.template.metadata.labels, message=Invalid value: map[string]string(nil): `selector` does not match template `labels`, reason=FieldValueInvalid, additionalProperties={})], group=apps, kind=StatefulSet, name=kubernetes-docker-build-and-deploy-statefulset, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=StatefulSet.apps "kubernetes-docker-build-and-deploy-statefulset" is invalid: spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Invalid, status=Failure, additionalProperties={}).
```

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AddStatefulSetResourceDecorator.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AddStatefulSetResourceDecorator.java
@@ -51,6 +51,8 @@ public class AddStatefulSetResourceDecorator extends ResourceProvidingDecorator<
                 .editOrNewTemplate()
                 .editOrNewSpec()
                 .endSpec()
+                .editOrNewMetadata()
+                .endMetadata()
                 .endTemplate();
 
         // defaults for:

--- a/integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KubernetesWithStatefulSetResourceTest.java
+++ b/integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KubernetesWithStatefulSetResourceTest.java
@@ -57,6 +57,9 @@ public class KubernetesWithStatefulSetResourceTest {
                             assertThat(podSpec.getTerminationGracePeriodSeconds()).isEqualTo(10);
                             assertThat(podSpec.getContainers()).allMatch(c -> APP_NAME.equals(c.getName()));
                         });
+                        assertThat(t.getMetadata()).satisfies(metadata -> {
+                            assertThat(metadata.getLabels()).containsEntry("app.kubernetes.io/name", APP_NAME);
+                        });
                     });
                     assertThat(statefulSetSpec.getSelector()).satisfies(ls -> {
                         assertThat(ls.getMatchLabels()).containsEntry("app.kubernetes.io/name", APP_NAME);

